### ISSUE TITLE
Explanation of the work

### ISSUE THREAD

**Issue by Dulyaaa** [parthdasawant/Segmentation-of-Sclera#4]
Hi @parthdasawant, this is a very nice✨ project work and seems like it's very useful. Since I'm a beginner some points are not clear for me. Can you explain little bit (step by step) what you have done here. 

Apparently, I have following questions: 
1. Which machine learning models that you have used?
2. What that ML model do with real eye image and masked image of eye? (Like how they are predict)

**Comment by parthdasawant** [comment]
Hello @Dulyaaa, This is Image Segmentation model build using UNet (Convolutional Neural Network). Image segmentation is the task of clustering parts of an image together that belong to the same object class. In this project we have done Semantic Segmentation for the Sclera(The white outer layer of the eyeball). 

Answers for your questions:

1. Actually this comes under Deep Learning since we have used CNN(i.e. UNet). The model was trained by us using SBVPI dataset.
2. The model is responsible for the generation of the mask for given image. Which is going to be used to extract the feature from that input image. Also the model was train using Supervised Learning therefore in that the mask images are the Ground Truth for respective training images.

    

The overall workflow was,
![image](https://user-images.githubusercontent.com/80618762/186492658-1a5690fd-21c8-4dce-9ec6-d40a9868050a.png)

**Comment by Dulyaaa** [comment]
> Hello @Dulyaaa, This is Image Segmentation model build using UNet (Convolutional Neural Network). Image segmentation is the task of clustering parts of an image together that belong to the same object class. In this project we have done Semantic Segmentation for the Sclera(The white outer layer of the eyeball).
> 
> Answers for your questions:
> 
> 1. Actually this comes under Deep Learning since we have used CNN(i.e. UNet). The model was trained by us using SBVPI dataset.
> 2. The model is responsible for the generation of the mask for given image. Which is going to be used to extract the feature from that input image. Also the model was train using Supervised Learning therefore in that the mask images are the Ground Truth for respective training images.
> 
> The overall workflow was, ![image](https://user-images.githubusercontent.com/80618762/186492658-1a5690fd-21c8-4dce-9ec6-d40a9868050a.png)

oh.. great @parthdasawant!!!
Actually, I just want to do the same procedure for my project. But the eye sclera area is fully red like it is not white like below, in my case. 

#### Input
![image](https://user-images.githubusercontent.com/57215584/187136547-f1697221-2ccc-4f37-9c94-576d9884aaef.png)

#### Output
![image](https://user-images.githubusercontent.com/57215584/187137875-12d40bba-e411-4fcf-8600-8168135005b4.png)

I tried the same thing that you have done. But seems like it's not working since I didn't get the correct output. What do you think about it? Hope, I didn't miss any thing. Do you have any recommendations or could you please point out me where 's the wrong.

And again thank you so much.

**Comment by parthdasawant** [comment]
@Dulyaaa if you have followed same procedure then you get the Segmented Mask(2nd pic from the workflow) as output from the prediction from the model. That Segmented Mask can be used for the BITWISE AND with your output image that you have posted in your comment.

`Final Output = cv2.bitwise_and(Output, Segmented Mask)`

**Comment by Dulyaaa** [comment]
Hiya @parthdasawant, Apparently, I will try the workflow again as you said. (Feels like I did a mistake somewhere else👀). So I will close the issue from here. (Hope, I will able to get back you if there is something to clarify)

Thank you soo much for your help. Great Work!! Keep it up!! Good Luck!!
